chain/ethereum/src/runtime/runtime_adapter.rs

```diff
@@ -103,7 +103,8 @@ fn ethereum_call(
     abis: &[Arc<MappingABI>],
     eth_call_gas: Option<u32>,
 ) -> Result<AscEnumArray<EthereumValueKind>, HostExportError> {
-    ctx.gas.consume_host_fn(ETHEREUM_CALL)?;
+    ctx.gas
+        .consume_host_fn_with_metrics(ETHEREUM_CALL, "ethereum_call")?;
```

Comment:
There is a macro crate `function_name!` https://docs.rs/function_name/latest/function_name/

Which can be potentially used here instead of hardcoding then name. Thoughts?

Reply:
The nicest would be an enum for the gas ops, but you don't need to spend time refactoring that right now.